packages/core/src/internal/execution/future-processor/future-processor.ts

```diff
@@ -96,6 +96,13 @@ export class FutureProcessor {
         exState !== undefined,
         `Invalid initialization message for future ${future.id}: it didn't create its execution state`
       );
+
+      if (
+        initMessage.type ===
+        JournalMessageType.CONTRACT_AT_EXECUTION_STATE_INITIALIZE
+      ) {
+        await this._recordDeployedAddressIfNeeded(initMessage);
```

Comment:
We need this check because `_recordDeployedAddressIfNeeded` never gets called for `contractAt` messages without this line, and we don't want to call it for every initialization message.

Reply:
I agree we keep the call to `_recoredDeployedAddressIfNeeded` here. But I think the purpose of the `_recoredDeployedAddressIfNeeded` abstraction is to centralise the rules of which messages are recorded to deployed addresses. That makes the `CONTRACT_AT_EXECUTION_STATE_INITIALIZE` check here a mirror.

You are right that removing it means we will end up calling `_recoredDeployedAddressIfNeeded` for every message, but the alternative is remembering to check that this still makes sense whenever we code `_recoredDeployedAddressIfNeeded` changes. My instinct is that the additional overhead of in-memory conditionals are going to get drowned out by our interactions with the node - so prefer code simplicity.

This is **not** a hill I am willing to die on though, happy to keep it.

Reply:
I think I'm just confused about how you envision this code working if we remove this line. I originally didn't have this line included, but without it, `_recoredDeployedAddressIfNeeded` never gets called anywhere for `CONTRACT_AT_EXECUTION_STATE_INITIALIZE` journal messages.